oracle/oracle.go

```diff
@@ -24,252 +27,131 @@ type Oracle interface {
-	for _, priceProvider := range o.providers {
-		o.fetchPrices(priceProvider)
-	}
-
-	o.logger.Debug("oracle fetched prices from providers")
-
-	// Compute aggregated prices and update the oracle.
-	o.priceAggregator.AggregatePrices()
-	o.setLastSyncTime(time.Now().UTC())
-
-	// update the last sync time
-	o.metrics.AddTick()
-
-	o.logger.Info("oracle updated prices", zap.Time("last_sync", o.GetLastSyncTime()), zap.Int("num_prices", len(o.GetPrices())))
+	return o.priceProviders
 }
 
-// fetchPrices retrieves the latest prices from a given provider and updates the aggregator
-// iff the price age is less than the update interval.
-func (o *OracleImpl) fetchPrices(provider *types.PriceProvider) {
-	defer func() {
-		if r := recover(); r != nil {
-			o.logger.Error("provider panicked", zap.Error(fmt.Errorf("%v", r)))
-		}
-	}()
-
-	if !provider.IsRunning() {
-		o.logger.Debug(
-			"provider is not running",
-			zap.String("provider", provider.Name()),
-		)
-
-		return
-	}
-
-	o.logger.Debug(
-		"retrieving prices",
-		zap.String("provider", provider.Name()),
-		zap.String("data handler type",
-			string(provider.Type())),
-	)
-
-	// Fetch and set prices from the provider.
-	prices := provider.GetData()
-	if prices == nil {
-		o.logger.Debug(
-			"provider returned nil prices",
-			zap.String("provider", provider.Name()),
-			zap.String("data handler type", string(provider.Type())),
-		)
-
-		return
-	}
-
-	timeFilteredPrices := make(types.Prices)
-	for pair, result := range prices {
-		// If the price is older than the maxCacheAge, skip it.
-		diff := time.Now().UTC().Sub(result.Timestamp)
-		if diff > o.maxCacheAge {
-			o.logger.Debug(
-				"skipping price",
-				zap.String("provider", provider.Name()),
-				zap.String("data handler type", string(provider.Type())),
-				zap.String("pair", pair.String()),
-				zap.Duration("diff", diff),
-			)
-
-			continue
-		}
+// GetMarketMap returns the market map.
+func (o *OracleImpl) GetMarketMap() mmtypes.MarketMap {
+	o.mut.Lock()
+	defer o.mut.Unlock()
 
-		o.logger.Debug(
-			"adding price",
-			zap.String("provider", provider.Name()),
-			zap.String("data handler type", string(provider.Type())),
-			zap.String("pair", pair.String()),
-			zap.String("price", result.Value.String()),
-			zap.Duration("diff", diff),
-		)
-		timeFilteredPrices[pair.GetOffChainTicker()] = result.Value
-	}
+	return o.marketMap
+}
 
-	o.logger.Debug("provider returned prices",
-		zap.String("provider", provider.Name()),
-		zap.String("data handler type", string(provider.Type())),
-		zap.Int("prices", len(prices)),
-	)
-	o.priceAggregator.SetProviderPrices(provider.Name(), timeFilteredPrices)
+func (o *OracleImpl) GetMarketMapProvider() *mmclienttypes.MarketMapProvider {
+	return o.mmProvider
 }
 
-// GetLastSyncTime returns the last time the oracle successfully updated prices.
 func (o *OracleImpl) GetLastSyncTime() time.Time {
-	o.mtx.RLock()
-	defer o.mtx.RUnlock()
-
 	return o.lastPriceSync
```

Comment:
Is there a reason we removed the read lock on this?

Reply:
I think we still need this.

Reply:
kek i dont know why this got removed, added back here [add read lock to GetLastSyncTime and nil check aggregator](https://github.com/skip-mev/slinky/pull/503/commits/8cb65abfcb2f0d9f70bfcc91acb30b11a7416198)